wallet-new/server/Cardano/Wallet/Server/Plugins.hs

```diff
@@ -121,9 +122,30 @@ legacyWalletBackend WalletBackendParams {..} ntpStatus =
       return $ withMiddleware walletRunMode app
 
     exceptionHandler :: SomeException -> Response
-    exceptionHandler _ =
-        responseLBS badRequest400 [(hContentType, "application/json")] .
-            encode $ UnkownError "Something went wrong."
+    exceptionHandler se =
+        case asum [handleV1Errors se, handleV0Errors se] of
```

Comment:
I guess:
```
let Just response = asum [handleV1Errors se, handleV0Errors se, Just $ handleGenericError se]
in response
``` 
Is less readable? :man_shrugging:

Reply:
That's also a possibility -- the snag would be that accidentally removing the last `Just $` during a refactoring and leaving the explicit pattern match intact would lead to a runtime exception. This version is shotgun-proof (as in preventing one to shoot himself in the foot 😁 ).

Reply:
While I agree that your version is more readable, removing the last `Just $` would result in a compile error. I think.

Reply:
Yup, you're right. I mean, your version is perfectly acceptable and I might have written that myself just as easily -- it's just that in the past I've got burned from explicit pattern matches like the above that these days tend to be more verbose at the expense of terseness. Both are perfectly fine 😉

Reply:
IMO we should make partial pattern matches a compile-error -- I'm pretty confident that we need `-fwarn-missing-unipatterns` or similar to get that in addition to `-Wall`.